File: qb-engineer-ui/src/app/features/parts/models/create-bom-entry-request.model.ts
import { BOMSourceType } from './bom-source-type.type';

export interface CreateBOMEntryRequest {
  childPartId: number;
  quantity: number;
  referenceDesignator?: string;
  sourceType: BOMSourceType;
  leadTimeDays?: number;
  notes?: string;
}

File: qb-engineer-server/qb-engineer.core/Models/CreateBOMEntryRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record CreateBOMEntryRequestModel(
    int ChildPartId,
    decimal Quantity,
    string? ReferenceDesignator,
    BOMSourceType SourceType,
    int? LeadTimeDays,
    string? Notes);
